### ISSUE TITLE
Enhancement:  Forbid optional chaining in `TaggedTemplateExpression`

### ISSUE THREAD

**Issue by fisker** [typescript-eslint/typescript-eslint#11352]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

```js
a?.b``
```

This should be invalid.

[Playground](https://typescript-eslint.io/play#ts=5.8.2&showAST=es&fileType=.tsx&code=IYfgdARgBlQ&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

**Comment by asdf93074** [comment]
Doesn't this happen already? From the playground:

<img width="1532" height="205" alt="typescript-eslint playground showing the string 'a?.b``' gives an error: Tagged template expressions are not permitted in an optional chain." src="https://github.com/user-attachments/assets/c9d3997b-e874-4f71-950b-edae4a18e435" />

or is the expectation that a parse error is thrown and ESTree doesn't get built fully?

**Comment by Josh-Cena** [comment]
It's a TypeScript error, which only shows up when you invoke the type checker. Our parser doesn't invoke the type checker, so an AST is indeed produced (which you can view in the ESTree tab).

**Comment by asdf93074** [comment]
> It's a TypeScript error, which only shows up when you invoke the type checker. Our parser doesn't invoke the type checker, so an AST is indeed produced (which you can view in the ESTree tab).

So is the expectation that the AST shouldn't be produced and a parser error should be thrown? I have a PR for it which does this but I'm just trying to be clear on the expectations.

<img width="1584" height="337" alt="typescript-eslint playground showing an internal error (parser error) on 'a?.b``' from typescript-eslint." src="https://github.com/user-attachments/assets/e81a011d-c5f0-4f3c-a04f-3d558be64e5d" />

**Comment by Josh-Cena** [comment]
It shouldn't be a hard error, though. You can use `this.#throwError`. Take a look at https://github.com/typescript-eslint/typescript-eslint/pull/9225 for example.

**Comment by asdf93074** [comment]
I was using `this.#throwUnlessAllowInvalidAST` which uses throwError under the the hood but anyways, I've raised a PR so we can discuss it there.
